### PR TITLE
Modify authorization state machine

### PR DESCRIPTION
**Pull request by peterschwarz**
This change require tracking the states more closely. The states of remote acceptance and remote identified have been added, so that these states can be properly transitioned through to the final Authorized state.  

It also allows for the use of the state value to indicate whether or not authorization is complete, as well as hold the identity.

### CODE REVIEW

File: libsplinter/src/network/auth/handlers.rs [hunk 1]
```diff
@@ -120,10 +128,31 @@ impl Handler for AuthorizedHandler {
...
+        debug!(
+            "Received authorize message from {}",
+            context.source_connection_id()
+        );
+        match self.auth_manager.next_state(
+            context.source_connection_id(),
+            AuthorizationAction::RemoteAuthorizing,
+        ) {
+            Err(err) => {
+                debug!(
```
Comment (vaporos) [inline libsplinter/src/network/auth/handlers.rs:140]: should this be warn?

Comment (vaporos) [reply libsplinter/src/network/auth/handlers.rs:140]: should this cause the connection to close?

Comment (peterschwarz) [reply libsplinter/src/network/auth/handlers.rs:140]: The connection manager will get the `Unauthorized` result, and close the connection.

Comment (peterschwarz) [reply libsplinter/src/network/auth/handlers.rs:140]: Possibly those should all be `warn`. I'll add a commit for that.

**Comment by peterschwarz** [comment]
Added a commit for the log levels, and specificity.

**Pull request merged**